frontend/src/plugin/runPlugin.ts

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2025 The Kubernetes Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ *
+ * @param pluginPath path to plugin
+ * @param packageName name of package
+ * @param packageVersion version of package
+ * @param permissionSecrets permission secrets are keyed by permission name, valued by secret
+ * @param handleError call back when an execution error occurs in the plugin
+ * @param getAllowedPermissions call back which returns only allowed permission secrets for plugin
+ * @param getArgValues call back which returns the argument names and values for the plugin
+ * @param privateFunction is a non global private copy of Function
+ *
+ */
+export type runPluginProps = [
+  /** path to plugin */
+  source: string,
+  /** name of package */
+  packageName: string,
+  /** version of package */
+  packageVersion: string,
+  /** call back when an execution error occurs in the plugin */
+  handleError: (error: unknown, packageName: string, packageVersion: string) => void,
+  /** is a non global private copy of Function */
+  PrivateFunction: typeof Function,
+  /** The argument names to be given to the plugin */
+  args: string[],
+  /** The values for the arguments to be given to the plugin */
+  values: unknown[],
+  /** A private copy of runPlugin */
+  internalRunPlugin: typeof runPlugin,
+  /** A private copy of console.error */
+  consoleError: typeof console.error
+];
+
+/**
+ * Runs a plugin by executing the source code in the global scope.
```

Comment:
This doc is not accurate or not in the right function I believe.

Reply:
fixed